weaviate/collections/base.py

```diff
@@ -20,8 +22,30 @@
 
 
 class _CollectionBase:
-    def __init__(self, name: str) -> None:
+    def __init__(self, connection: Connection, name: str) -> None:
+        self._connection = connection
         self.name = _capitalize_first_letter(name)
+        self.__cluster = _Cluster(connection)
+
+    def shards(self) -> List[Shard]:
+        """
+        Get the statuses of all the shards of this collection.
+
+        Returns:
+            The list of shards belonging to this collection.
+
+        Raises
+            `requests.ConnectionError`
+                If the network connection to weaviate fails.
+            `weaviate.UnexpectedStatusCodeException`
+                If weaviate reports a none OK status.
+            `weaviate.EmptyResponseException`
+                If the response is empty.
+        """
+        shards: List[Shard] = []
+        for node in self.__cluster.nodes(self.name, output="verbose"):
+            shards.extend(node.shards)
```

Comment:
list comprehension? Should be a bit faster when there are a lot of shards

Reply:
wow didn't know you could do this, will change!